packages/auth/src/grant/routes.ts

```diff
@@ -179,6 +180,65 @@ function isContinuableGrant(grant: Grant): boolean {
   return !isRejectedGrant(grant) && !isRevokedGrant(grant)
 }
 
+function isGrantStillWaiting(grant: Grant): boolean {
+  if (!grant.wait) return false
+  const grantWaitTime = grant.createdAt.getTime() + grant.wait * 1000
+  const currentTime = Date.now()
+
+  return currentTime < grantWaitTime
+}
+
+async function pollGrantContinuation(
+  deps: ServiceDependencies,
+  ctx: ContinueContext,
+  continueId: string,
+  continueToken: string
+): Promise<void> {
+  const { config, grantService, accessService, accessTokenService } = deps
+
+  const grant = await grantService.getByContinue(continueId, continueToken)
+  if (!grant) {
+    ctx.throw(404, { error: 'unknown_request' })
+  }
+
+  if (isGrantStillWaiting(grant)) {
+    ctx.throw(401, { error: 'too_fast' })
```

Comment:
Can we be more descriptive in those error messages?

Reply:
I think @njlie has been following https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol-15#response-error

I think we should keep these errors by putting them in the `code` field (as defined in the spec), while adding the `description` field